lib/src/screens/receive/lightning_invoice_page.dart

```diff
@@ -0,0 +1,346 @@
+  @override
+  Widget middle(BuildContext context) => PresentReceiveOptionPicker(
+      receiveOptionViewModel: receiveOptionViewModel, color: titleColor(context));
+
+  @override
+  Widget trailing(BuildContext context) => TrailButton(
+      caption: S.of(context).clear,
+      onPressed: () {
+        _formKey.currentState?.reset();
+      });
+
+  Future<bool> _onNavigateBack(BuildContext context) async {
+    onClose(context);
+    return false;
+  }
+
+  @override
+  Widget body(BuildContext context) {
+    WidgetsBinding.instance.addPostFrameCallback((_) => _setReactions(context));
+
+    return WillPopScope(
+      onWillPop: () => _onNavigateBack(context),
+      child: KeyboardActions(
+        disableScroll: true,
+        config: KeyboardActionsConfig(
+            keyboardActionsPlatform: KeyboardActionsPlatform.IOS,
+            keyboardBarColor: Theme.of(context).extension<KeyboardTheme>()!.keyboardBarColor,
+            nextFocus: false,
+            actions: [
+              KeyboardActionsItem(
+                focusNode: _amountFocusNode,
+                toolbarButtons: [(_) => KeyboardDoneButton()],
+              ),
+            ]),
+        child: Container(
+          color: Theme.of(context).colorScheme.background,
+          child: ScrollableWithBottomSection(
+            contentPadding: EdgeInsets.only(bottom: 24),
+            content: Container(
+              decoration: responsiveLayoutUtil.shouldRenderMobileUI
+                  ? BoxDecoration(
+                      borderRadius: BorderRadius.only(
+                          bottomLeft: Radius.circular(24), bottomRight: Radius.circular(24)),
+                      gradient: LinearGradient(
+                        colors: [
+                          Theme.of(context)
+                              .extension<ExchangePageTheme>()!
+                              .firstGradientTopPanelColor,
+                          Theme.of(context)
+                              .extension<ExchangePageTheme>()!
+                              .secondGradientTopPanelColor,
+                        ],
+                        begin: Alignment.topLeft,
+                        end: Alignment.bottomRight,
+                      ),
+                    )
+                  : null,
+              child: Observer(builder: (_) {
+                return Padding(
+                  padding: EdgeInsets.fromLTRB(24, 120, 24, 0),
+                  child: LightningInvoiceForm(
+                    descriptionController: _descriptionController,
+                    amountController: _amountController,
+                    depositAmountFocus: _amountFocusNode,
+                    formKey: _formKey,
+                    lightningInvoicePageViewModel: lightningInvoicePageViewModel,
+                  ),
+                );
+              }),
+            ),
+            bottomSectionPadding: EdgeInsets.only(left: 24, right: 24, bottom: 24),
+            bottomSection: Observer(builder: (_) {
```

Comment:
same for this one, probably just wrap `LoadingPrimaryButton` with Observer not the whole `bottomSection`, but again, make sure everything is behaving correctly if you will remove them